crates/core/src/credits.rs

```diff
@@ -0,0 +1,113 @@
+use std::{collections::HashMap, io::prelude::*, path::PathBuf};
+
+use hub_core_schemas::{CreditsEvent, CreditsEventKey};
+
+use crate::{prelude::*, producer, util::DebugShim};
+
+impl producer::Message for CreditsEvent {
+    type Key = CreditsEventKey;
+}
+
+/// Service startup configuration for charging credits and reading the credit
+/// sheet
+#[derive(Debug)]
+pub struct Config {
+    pub(crate) credit_sheet: PathBuf,
+    pub(crate) config: DebugShim<rdkafka::ClientConfig>,
+}
+
+impl Config {
+    /// Construct a new credits producer client from this config instance
+    #[inline]
+    pub async fn build<I: LineItem>(self) -> Result<CreditsClient<I>> {
+        CreditsClient::new(self).await
+    }
+}
+
+/// A client for producing credit deduction line items
+#[derive(Debug)]
```

Comment:
```suggestion
#[derive(Debug, Clone)]
```
need the credits client to implement Clone